recipes/libvpx/all/conanfile.py

```diff
@@ -0,0 +1,151 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class LibVPXConan(ConanFile):
+    name = "libvpx"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.webmproject.org/code"
+    description = "WebM VP8/VP9 Codec SDK"
+    topics = ("conan", "vpx", "codec", "web", "VP8", "VP9")
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'fPIC': True}
+
+    _source_subfolder = "source_subfolder"
+
+
+    def configure(self):
+        if self.settings.os == 'Windows' and self.options.shared:
+            raise ConanInvalidConfiguration('Windows shared builds are not supported')
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires('yasm/1.3.0')
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires('msys2/20200517')
+
+    def source(self):                                                                                    
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _fix_sources(self):
+        gen_vcxproj = os.path.join(self._source_subfolder, 'build', 'make', 'gen_msvs_vcxproj.sh')
```

Comment:
Better to have a patch for this non-trivial replacement.

Reply:
okay, let's do it

Reply:
moved to patch